src/models/dynamic_generator.jl

```diff
@@ -97,32 +84,72 @@ function DynamicGenerator(
 end
 
 function DynamicGenerator(;
-    static_injector::Generator,
+    name::Union{String, Generator},
```

Comment:
@jd-lara We need to discuss the workflows here.  Does the user pass the generator, its name, or either?